From 027b2e172fcd29db409bd41ca1925a32d02c14c0 Mon Sep 17 00:00:00 2001
From: Sven Nierlein <sven@nierlein.de>
Date: Mon, 10 Dec 2018 17:46:39 +0100
Subject: [PATCH] fix query handler not returning command response

there was a missing newline which prevented the query handler
from returning the errors for commands.
add a test case to ensure this does not fail again.
---
 src/naemon/query-handler.c |  2 +-
 tests.mk                   |  6 +++
 tests/test-query-handler.c | 92 ++++++++++++++++++++++++++++++++++++++
 3 files changed, 99 insertions(+), 1 deletion(-)
 create mode 100644 tests/test-query-handler.c

diff --git a/src/naemon/query-handler.c b/src/naemon/query-handler.c
index dbc20ac3..2dc29a7e 100644
--- a/src/naemon/query-handler.c
+++ b/src/naemon/query-handler.c
@@ -353,7 +353,7 @@ static int qh_command(int sd, char *buf, unsigned int len)
 			if (res == OK) {
 				return 200;
 			} else {
-				nsock_printf_nul(sd, "400: %s", error->message);
+				nsock_printf_nul(sd, "400: %s\n", error->message);
 				g_clear_error(&error);
 				return 0;
 			}
